stub/include/neurala/meta/detail/typeName.h

```diff
@@ -32,7 +34,7 @@ namespace neurala
  * @warning This is not a null-terminated string, so @ref size() is the only function that
  *          can convey it's length properly.
  */
-class StaticString
+class NEURALA_PUBLIC StaticString
```

Comment:
I even exported "detail" types, maybe the upcoming tickets should judge if these belong? Is it not reasonable to automatically consider anything used in this stub worth exporting and move the files out of the `detail` dirs?

Reply:
I am surprised typetraits need to be exported.